### ISSUE TITLE
Does Yelp Fusion support IPv6?

### ISSUE THREAD

**Issue by RaghavMangrola** [Yelp/yelp-fusion#142]
Apologies if this is a silly question, I'm submitting my app update - https://github.com/RaghavMangrola/TacoButton 

It's being rejected by apple because it crashes whenever I make a search request on an IPv6 connection

**Comment by mittonk** [comment]
@RaghavMangrola : Not silly, worth digging into.

Can you provide any extra info about the crash?  (Don't share anything Apple wouldn't want you to.)

**Comment by RaghavMangrola** [comment]
This was their message to me

---

Performance - 2.1


Your app crashed on iPad and iPhone running iOS 10.1.1 connected to an IPv6 network when we tap the “Find Tacos!” button.

This occurred when your app was used:

- On Wi-Fi

We have attached detailed crash logs to help troubleshoot this issue.

Next Steps

Please revise your app and test it on a device while connected to an IPv6 network (all apps must support IPv6) to ensure that it runs as expected.

---
 
And some crash logs provided by them

[attachment-3125287472890141648crashlog-9D8DC48C-0748-45C1-AED8-8701836D2108.txt](https://github.com/Yelp/yelp-fusion/files/664467/attachment-3125287472890141648crashlog-9D8DC48C-0748-45C1-AED8-8701836D2108.txt)
[attachment-4892799584895054290crashlog-1FAA9349-6313-427A-BB9B-DAD7A799939F.txt](https://github.com/Yelp/yelp-fusion/files/664468/attachment-4892799584895054290crashlog-1FAA9349-6313-427A-BB9B-DAD7A799939F.txt)
[attachment-6536067616517151560crashlog-18AD0C5E-5E88-4EA7-BE73-5EC970198D18.txt](https://github.com/Yelp/yelp-fusion/files/664469/attachment-6536067616517151560crashlog-18AD0C5E-5E88-4EA7-BE73-5EC970198D18.txt)

---

**Comment by mittonk** [comment]
You might be able to reproducer the crash locally using a NAT64 test network:
* http://stackoverflow.com/questions/39418928/ipv6-swift-reachability
* https://developer.apple.com/library/content/documentation/NetworkingInternetWeb/Conceptual/NetworkingOverview/UnderstandingandPreparingfortheIPv6Transition/UnderstandingandPreparingfortheIPv6Transition.html#//apple_ref/doc/uid/TP40010220-CH213-SW16

We're looking into the server side of things.

**Comment by tzuhanwu** [comment]
Hello @RaghavMangrola,
api.yelp.com indeed doesn't support IPv6, but that's not what's causing your crashes or App Store rejections. There's no requirement from Apple for supporting IPv6 on the server side, usually network carriers do the IP-addresses conversion in between IPv6 client and IPv4 server. You can find more information here: [Supporting IPv6 DNS64/NAT64 Networks](https://developer.apple.com/library/content/documentation/NetworkingInternetWeb/Conceptual/NetworkingOverview/UnderstandingandPreparingfortheIPv6Transition/UnderstandingandPreparingfortheIPv6Transition.html#//apple_ref/doc/uid/TP40010220-CH213-SW1).

Here are some ideas that might help you identify the problem:
1. Are you doing something like checking the connection to 0.0.0.0?
2. Are you scanning the binary for strings with ipv4 addresses?

**Comment by mittonk** [comment]
(Apple scanning the binary for IPv4 addresses wouldn't explain these particular stacktraces, I don't think that's what we're chasing.)

**Comment by RaghavMangrola** [comment]
I went to Apple, and I explained to them that the only network request I was making was to the Yelp Fusion API using NSURLSession and that I wasn't using any literal IP Addresses. I sent them a link to this thread to see your findings. They re-reviewed my app this morning, found that it didn't crash, and approved my new update this morning.

Thanks for taking your time to help me out, I really appreciate all the effort you both took to help me get my app approved.

Hope you have a great 2017!
